app/IATI/Services/Publisher/PublisherService.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\IATI\Services\Publisher;
+
+use App\IATI\Services\Workflow\RegistryApiHandler;
+use App\IATI\Traits\RegistryApiInvoker;
+use GuzzleHttp\Exception\ClientException;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Storage;
+
+/**
+ * Class PublisherService.
+ */
+class PublisherService extends RegistryApiHandler
+{
+    use RegistryApiInvoker;
+
+    /**
+     * @var Model|null
+     */
+    protected $activityPublished = null;
+
+    /**
+     * Publishes the activity xml file to the IATI registry.
+     *
+     * @param $registryInfo
+     * @param $activityPublished
+     * @param $organization
+     *
+     * @return void
+     */
+    public function publishFile($registryInfo, $activityPublished, $organization): void
+    {
+        $this->setFile($activityPublished);
+        $this->init(env('IATI_API_ENDPOINT'), Arr::get($registryInfo, 'api_token', ''))
+             ->setPublisher(Arr::get($registryInfo, 'publisher_id', ''));
+        $this->searchForPublisher($this->publisherId);
+        $this->publishToRegistry($organization, $activityPublished->filename);
+    }
+
+    /**
+     * Set the file attribute.
+     *
+     * @param $activityPublished
+     */
+    protected function setFile($activityPublished)
+    {
+        $this->activityPublished = $activityPublished;
+    }
+
+    /**
+     * Publish File to the IATI Registry.
+     *
+     * @param $organization
+     * @param $filename
+     *
+     * @return void
+     */
+    protected function publishToRegistry($organization, $filename): void
+    {
+        $data = $this->generatePayload($organization, $filename);
```

Comment:
$data = $this->generatePayload($organization, $filename);

    if ($this->isPackageAvailable($this->extractPackage($filename), $this->apiKey)) {
        $this->client->package_update($data);
    } else {
        $this->client->package_create($data);
    }